src/set.js

```diff
@@ -0,0 +1,99 @@
+export default class Set {
+  constructor() {
+    this.length = 0
+    this.elements = []
+  }
+
+  add = (value) => {
+    if(!this.contains(value)){
+      this.elements.push(value)
+      this.length += 1
+    }
+  }
+
+  isEmpty = () => this.length === 0
+
+  contains = (value) => {
+    for (let i = 0; i < this.length; i++) {
+      if (this.elements[i] === value) {
+        return true
+      }
+    }
+    return false
+  }
+
+  remove = (value) => {
+    if(this.contains(value)){
+      let removeIndex = 0
+      for(let i = 0; i < this.length; i++){
+        if(this.elements[i] === value){
+          removeIndex = i
+        }
+      }
+      this.elements.splice(removeIndex, 1)
+      this.length -= 1
+    }
+  }
+
+  forEach = passedInFunction => {
+    const setSize = new Number(this.length)
+    for (let i = 0; i < setSize; i++) {
+      passedInFunction(this.elements[i])
+    }
+  }
+
+  size = () => this.length
+
+  union = (otherSet) => {
+    let newSet = new Set()
+    newSet = this
+    for(let i = 0; i < otherSet.length; i++) {
+      if(!newSet.contains(otherSet.elements[i])) {
+        newSet.add(otherSet.elements[i])
+      }
+    }
+    return newSet
+  }
+
+  intersect = (otherSet) => {
+    let newSet = new Set()
+    newSet = this
+    for(let i = 0; i < this.length; i++) {
+      if(!newSet.contains(otherSet.elements[i])) {
+        newSet.remove(otherSet.elements[i])
+      }
+    }
+    return newSet
+  }
+
+  difference = (otherSet) => {
+    let newSet = new Set()
+
+    for(let i = 0; i < this.length; i++) {
+      if(!otherSet.contains(this.elements[i])) {
+        newSet.add(this.elements[i])
+        }
+      }
+    return newSet
+  }
+
+  isSubset = (otherSet) => {
+    let count = 0
+    for(let i = 0; i < this.size(); i++){
+      if(otherSet.contains(this.elements[i])){
+        count += 1
+      }
+    }
+    if(count === this.size()){
```

Comment:
When I see the pattern:
```js
if( booleanExpression ) {
  return true
} else {
  return false
}
```
I replace it with:
```js
return booleanExpression
```
Here, you could just `return count === this.size()`